.github/actions/trun/main.ts

```diff
@@ -0,0 +1,89 @@
+import * as core from "./deps/actions/core.ts"
+import PQueue from "./deps/pqueue.ts"
+import puppeteer from "./deps/puppeteer.ts"
+import { deferred } from "./deps/std/async.ts"
+import { parse } from "./deps/std/flags.ts"
+import * as path from "./deps/std/path.ts"
+import { run, runWithBrowser, runWithDeno } from "./run.ts"
+
+const flags = parse(Deno.args, {
+  string: ["dir", "concurrency", "ignore", "import-map", "filter"],
+  boolean: ["browser"],
+  default: { ignore: ".ignore" },
+})
+
+const { ignore, dir, browser } = flags
+const importMap = flags["import-map"]
+const concurrency = flags.concurrency ? parseInt(flags.concurrency) : Infinity
+if (!dir) {
+  throw new Error("dir flag is required")
+}
+
+const filter = flags.filter ? new Set(flags.filter.split(",")) : new Set()
+
+const ignoreFile = await Deno.readTextFile(path.join(dir, ignore))
+const ignoredFiles = new Set(ignoreFile.split("\n"))
+
+const sourceFileNames = Array.from(Deno.readDirSync(dir))
+  .filter((e) =>
+    e.name.match(/^.*\.ts$/g)
+    && e.isFile
+    && !ignoredFiles.has(e.name)
+    && ((filter.size > 0 && filter.has(e.name)) || filter.size === 0)
+  )
+  .map((f) => f.name)
+
+const shutdownTasks = new PQueue()
```

Comment:
Can you use an `AbortController` instead of this `deferred`/`PQueue` combination?